Add updated architecture diagrams (#499)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/con_stf-architecture.adoc

[role="_abstract"]
{Project} ({ProjectShort}) uses a client-server architecture, in which {OpenStack} ({OpenStackShort}) is the client and {OpenShift} is the server.

> {ProjectShort} is capable of dealing with both metrics and events information from {OpenStack} ({OpenStackShort}) but the default installation is focused on collection, transport, and storage of metrics information.

Author: @mickogeary I like where you're going here. I think it's still worth calling out that we're capable of delivering events via forwarding to a user-provisioned Elasticsearch component. Putting the focus on the default though at the start of the paragraph makes sense to me.

> {ProjectShort} is capable of dealing with both metrics and events information from {OpenStack} ({OpenStackShort}) but the default installation is focused on collection, transport, and storage of metrics information.

Reviewer: can have the Elasticsearch mention in the next line if that helps? "You can collect {OpenStackShort} events data, transport it with the message bus, and forward it to a user-provided Elasticsearch from the Smart Gateways, but this option is deprecated."

> {ProjectShort} is capable of dealing with both metrics and events information from {OpenStack} ({OpenStackShort}) but the default installation is focused on collection, transport, and storage of metrics information.

Author: Hrmmmm I like what you wrote there, but on second thought I wonder if actually focusing on the metrics and kind of not discussing events here makes sense, and instead leave that for the new events guide with the separate architecture overview.

I'm going to merge your original suggestion. Thanks!

> Event data can be enabled on {OpenStack}, transported via the message bus, and then forwarded to a user-provided Elasticsearch from the Smart Gateways, but this architecture is deprecated. The discussion in this guide will be focused on the default deployment configuration.

Reviewer: ```suggestion
Collecting, transporting, and storing {OpenStackShort} events data is possible, but deprecated.
```

Author: Hrmmm.... I feel like this might be a bit too _little_ verbosity, but I get that my original paragraph was _too verbose_ as well :)

Reviewer: Maybe: "You can collect {OpenStackShort} events data, transport it with the message bus, and forward it to a user-provided Elasticsearch from the Smart Gateways, but this option is deprecated."

Author: I like this one better.

// For more information about {ProjectShort} when configured with events, see <xref-TODO>. Tracked via https://issues.redhat.com/browse/STF-1552

{ProjectShort} consists of the following components:

* Data collection

> ** collectd: Collects infrastructure metrics and events on {OpenStackShort}.

Reviewer 2: Do we have a list of the collectd plugins we support? I remember seeing a question along these lines, it might be a good idea to refer to some other document with this information.

Author: Basically the answer is, "whatever we ship", which has been pretty loose.  Documented plugins are at https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/17.1/html/managing_overcloud_observability/collectd-plugins_assembly

[[osp-stf-overview]]
.Service Telemetry Framework architecture overview
image::363_OpenStack_STF_updates_0923_arch_2.png[Service Telemetry Framework architecture overview]

For client side metrics, collectd provides infrastructure metrics without project data, and Ceilometer provides {OpenStackShort} platform data based on projects or user workload. Both Ceilometer and collectd deliver data to Prometheus by using the {MessageBus} transport, delivering the data through the message bus. On the server side, a Golang application called the Smart Gateway takes the data stream from the bus and exposes it as a local scrape endpoint for Prometheus.

> If you plan to collect and store events, collectd and Ceilometer deliver event data to the server side by using the {MessageBus} transport. Another Smart Gateway forwards the data to a user-provided Elasticsearch datastore.

Reviewer: ```suggestion
When you collect and store events, collectd and Ceilometer deliver event data to the server side by using the {MessageBus} transport. Another Smart Gateway forwards the data to a user-provided Elasticsearch datastore.
```

[NOTE]
====

> In previous releases of {ProjectShort}, the Service Telemetry Operator would request instances of Elasticsearch from the Elastic Cloud on Kubernetes (ECK) Operator. {ProjectShort} now uses a forwarding model where events are forwarded from a Smart Gateway instance to a user-provided instance of Elasticsearch. The management of an Elasticsearch instance by Service Telemetry Operator is deprecated.

Reviewer: ```suggestion
In previous releases of {ProjectShort}, the Service Telemetry Operator requested instances of Elasticsearch from the Elastic Cloud on Kubernetes (ECK) Operator. {ProjectShort} now uses a forwarding model, where events are forwarded from a Smart Gateway instance to a user-provided instance of Elasticsearch. The management of an Elasticsearch instance by Service Telemetry Operator is deprecated.
```

> New ServiceTelemetry instances will use the `observabilityStrategy` of `use_redhat` which does not request Elasticsearch instances from ECK. Instances of ServiceTelemetry deployed prior to {ProjectShort} 1.5.3 use an observabilityStrategy of use_community, which matches the previous architecture. In the case where a user previously deployed an Elasticsearch instance with {ProjectShort},  the Service Telemetry Operator will automatically update the ServiceTelemetry custom resource object to have the observabilityStrategy set to use_community, and remain functioning the same as previous releases. For more information about observability strategies, see xref:observability-strategy-in-service-telemetry-framework_assembly-preparing-your-ocp-environment-for-stf[].

Reviewer: ```suggestion
In new `ServiceTelemetry`deployments, the `observabilityStrategy` parameter has a value of `use_redhat`, that does not request Elasticsearch instances from ECK. Deployments of `ServiceTelemetry` that are version {ProjectShort} 1.5.3 or older have the `observabilityStrategy` parameter set to `use_community`, which matches the previous architecture. If a user deployed an Elasticsearch instance with {ProjectShort},  the Service Telemetry Operator updates the `ServiceTelemetry` custom resource object to have the `observabilityStrategy` parameter set to `use_community`, and function similar to previous releases. For more information about observability strategies, see xref:observability-strategy-in-service-telemetry-framework_assembly-preparing-your-ocp-environment-for-stf[].
```

> In new `ServiceTelemetry`deployments, the `observabilityStrategy` parameter has a value of `use_redhat`, that does not request Elasticsearch instances from ECK. Deployments of `ServiceTelemetry` that are version {ProjectShort} 1.5.3 or older have the `observabilityStrategy` parameter set to `use_community`, which matches the previous architecture. If a user deployed an Elasticsearch instance with {ProjectShort},  the Service Telemetry Operator updates the `ServiceTelemetry` custom resource object to have the `observabilityStrategy` parameter set to `use_community`, and function similar to previous releases. For more information about observability strategies, see xref:observability-strategy-in-service-telemetry-framework_assembly-preparing-your-ocp-environment-for-stf[].

Author: ```suggestion
In new `ServiceTelemetry`deployments, the `observabilityStrategy` parameter has a value of `use_redhat`, that does not request Elasticsearch instances from ECK. Deployments of `ServiceTelemetry` that are version {ProjectShort} 1.5.3 or older have the `observabilityStrategy` parameter set to `use_community`, which matches the previous architecture. If a user deployed an Elasticsearch instance with {ProjectShort}, the Service Telemetry Operator updates the `ServiceTelemetry` custom resource object to have the `observabilityStrategy` parameter set to `use_community`, and functions similar to previous releases. For more information about observability strategies, see xref:observability-strategy-in-service-telemetry-framework_assembly-preparing-your-ocp-environment-for-stf[].
```

> A user can migrate to the use_redhat observability strategy. For more information, see link:https://access.redhat.com/articles/7011708[Migrating Service Telemetry Framework to fully supported operators].

Reviewer: ```suggestion
For more information about migration to the `use_redhat` observability strategy, see link:https://access.redhat.com/articles/7011708[Migrating Service Telemetry Framework to fully supported operators].
```
